tests/Nidup/Architool/Infrastructure/Filesystem/FolderReferenceUpdaterTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+use PHPUnit\Framework\TestCase;
+use org\bovigo\vfs\vfsStreamWrapper;
+use org\bovigo\vfs\vfsStream;
+use org\bovigo\vfs\vfsStreamDirectory;
+use Nidup\Architool\Infrastructure\Filesystem\FolderReferenceUpdater;
+use Nidup\Architool\Domain\Model\Folder;
+use Nidup\Architool\Domain\Model\Folder\FolderNamespace;
+
+class FolderReferenceUpdaterTest extends TestCase
+{
+
+    private $root;
+
+    public function setUp()
+    {
+        vfsStreamWrapper::register();
+
+        $rootStructure = [
+            'My' => [
+                'Project' => [
+                    'Dir' => [
+                        'features' => [],
+                        'tests' => [],
+                        'app' => [
+                            'config' => []
+                        ],
+                        'src' => [
+                            'Akeneo' => [
+                                'Component' => [
+                                    'Batch' => [],
+                                    'BatchQueue' => [],
+                                ],
+                                'Pim' => [
+                                    'Platform' => [
+                                        'Domain' => [
+                                            'Batch' => []
+                                        ]
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+            ]
+        ];
+
+        $this->root = vfsStream::setup('root', null, $rootStructure);
+    }
+
+    public function testNamespaceDeclarationIsUpdatedInPhp()
+    {
+        $filePath =  $this->root->url().'/My/Project/Dir/src/Akeneo/Component/Batch/MyClassInBatchComponent.php';
+        file_put_contents(
+            $filePath,
+            'namespace Akeneo\Component\Batch;'
+        );
+
+        $folder = new Folder(new FolderNamespace('Akeneo/Component/Batch'));
+        $folder->move(new FolderNamespace('Akeneo/Pim/Platform/Domain/Batch'));
+
+        $updater = new FolderReferenceUpdater($this->root->url().'/My/Project/Dir');
+
+        $updater->update($folder);
+
+        $this->assertEquals(
+            'namespace Akeneo\Pim\Platform\Domain\Batch;',
+            file_get_contents($filePath)
+        );
+    }
+
+    public function testDifferentNamespaceDeclarationWithSamePrefixIsNotUpdatedInPhp()
```

Comment:
I don't understand this test :(

Reply:
I test if a different namespace declaration with same prefix is not updated.